File: AD/广点通/GDT_iOS_SDK_4-2/GDTMobApp/GDTAdTypes/RewardVideoViewController.m
//
//  RewardVideoViewController.m
//  GDTMobApp
//
//  Created by royqpwang on 2018/9/5.
//  Copyright © 2018年 Tencent. All rights reserved.
//

#import "RewardVideoViewController.h"
#import "GDTRewardVideoAd.h"
#import "GDTAppDelegate.h"

@interface RewardVideoViewController () <GDTRewardedVideoAdDelegate>

@property (nonatomic, strong) GDTRewardVideoAd *rewardVideoAd;
@property (weak, nonatomic) IBOutlet UITextField *portraitPlacementIdTextField;
@property (weak, nonatomic) IBOutlet UITextField *placementIdTextField;
@property (weak, nonatomic) IBOutlet UILabel *statusLabel;
@property (nonatomic, assign) UIInterfaceOrientation supportOrientation;
@property (weak, nonatomic) IBOutlet UIButton *portraitButton;
@property (weak, nonatomic) IBOutlet UIButton *botnButton;

@end

@implementation RewardVideoViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
{
    [super touchesBegan:touches withEvent:event];
    [self.view endEditing:YES];
}

- (BOOL)shouldAutorotate
{
    return YES;
}

- (IBAction)loadAd:(id)sender {
    if (sender == self.portraitButton) {
        self.rewardVideoAd = [[GDTRewardVideoAd alloc] initWithAppId:kGDTMobSDKAppId placementId:self.portraitPlacementIdTextField.text];
    } else {
        self.rewardVideoAd = [[GDTRewardVideoAd alloc] initWithAppId:kGDTMobSDKAppId placementId:self.placementIdTextField.text];
    }
    self.rewardVideoAd.delegate = self;
    [self.rewardVideoAd loadAd];
}

- (IBAction)playVideo:(UIButton *)sender {
    
    if (self.rewardVideoAd.expiredTimestamp <= [[NSDate date] timeIntervalSince1970]) {
        self.statusLabel.text = @"广告已过期，请重新拉取";
        return;
    }
    if (!self.rewardVideoAd.isAdValid) {
        self.statusLabel.text = @"广告失效，请重新拉取";
        return;
    }
    [self.rewardVideoAd showAdFromRootViewController:self];
}

- (IBAction)changeOrientation:(UIButton *)sender {
    // 仅为方便调试提供的逻辑，应用接入流程中不需要程序设置方向
    if (UIInterfaceOrientationIsPortrait([UIApplication sharedApplication].statusBarOrientation)) {
        self.supportOrientation = UIInterfaceOrientationLandscapeRight;
    } else {
        self.supportOrientation = UIInterfaceOrientationPortrait;
    }
    [[UIDevice currentDevice] setValue:@(self.supportOrientation) forKey:@"orientation"];
}

#pragma mark - GDTRewardVideoAdDelegate
- (void)gdt_rewardVideoAdDidLoad:(GDTRewardVideoAd *)rewardedVideoAd
{
    self.statusLabel.text = @"广告数据加载成功";
}


- (void)gdt_rewardVideoAdVideoDidLoad:(GDTRewardVideoAd *)rewardedVideoAd
{
    self.statusLabel.text = @"视频文件加载成功";
}


- (void)gdt_rewardVideoAdWillVisible:(GDTRewardVideoAd *)rewardedVideoAd
{
    NSLog(@"视频播放页即将打开");
}

- (void)gdt_rewardVideoAdDidExposed:(GDTRewardVideoAd *)rewardedVideoAd
{
    NSLog(@"广告已曝光");
}

- (void)gdt_rewardVideoAdDidClose:(GDTRewardVideoAd *)rewardedVideoAd
{
    self.statusLabel.text = @"广告已关闭";
    NSLog(@"广告已关闭");
}


- (void)gdt_rewardVideoAdDidClicked:(GDTRewardVideoAd *)rewardedVideoAd
{
    NSLog(@"广告已点击");
}

- (void)gdt_rewardVideoAd:(GDTRewardVideoAd *)rewardedVideoAd didFailWithError:(NSError *)error
{
    
    if (error.code == 4014) {
        NSLog(@"请拉取到广告后再调用展示接口");
        self.statusLabel.text = @"请拉取到广告后再调用展示接口";
    } else if (error.code == 4016) {
        NSLog(@"应用方向与广告位支持方向不一致");
        self.statusLabel.text = @"应用方向与广告位支持方向不一致";
    } else if (error.code == 5012) {
        NSLog(@"广告已过期");
        self.statusLabel.text = @"广告已过期";
    } else if (error.code == 4015) {
        NSLog(@"广告已经播放过，请重新拉取");
        self.statusLabel.text = @"广告已经播放过，请重新拉取";
    } else if (error.code == 5002) {
        NSLog(@"视频下载失败");
        self.statusLabel.text = @"视频下载失败";
    } else if (error.code == 5003) {
        NSLog(@"视频播放失败");
        self.statusLabel.text = @"视频播放失败";
    } else if (error.code == 5004) {
        NSLog(@"没有合适的广告");
        self.statusLabel.text = @"没有合适的广告";
    } else if (error.code == 5013) {
        NSLog(@"请求太频繁，请稍后再试");
    }
    NSLog(@"ERROR: %@", error);
}

- (void)gdt_rewardVideoAdDidRewardEffective:(GDTRewardVideoAd *)rewardedVideoAd
{
    NSLog(@"播放达到激励条件");
}

- (void)gdt_rewardVideoAdDidPlayFinish:(GDTRewardVideoAd *)rewardedVideoAd
{
    NSLog(@"视频播放结束");
}

@end

File: AD/广点通/GDT_iOS_SDK_4.9.0/GDTMobSample-Swift/GDTAdTypes/RewardVideoViewController/RewardVideoViewController.swift
//
//  RewardVideoViewController.swift
//  GDTMobSample-Swift
//
//  Created by royqpwang on 2018/10/15.
//  Copyright © 2018年 Tencent. All rights reserved.
//

import UIKit

class RewardVideoViewController: UIViewController, GDTRewardedVideoAdDelegate {

    private var rewardVideoAd:GDTRewardVideoAd!
    @IBOutlet weak var statusLabel: UILabel!
    @IBOutlet weak var portraitButton: UIButton!
    @IBOutlet weak var botnButton: UIButton!
    @IBOutlet weak var portraitPlacementIdTextField: UITextField!
    @IBOutlet weak var placementIdTextField: UITextField!
    
    var supportOrientation: UIInterfaceOrientation!

    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        super.touchesBegan(touches, with: event)
        self.view.endEditing(true)
    }
    
    @IBAction func loadAd(_ sender: Any) {
        if sender as? UIButton == self.portraitButton {
            self.rewardVideoAd = GDTRewardVideoAd.init(appId: Constant.appID, placementId: self.portraitPlacementIdTextField.text)
        } else {
            self.rewardVideoAd = GDTRewardVideoAd.init(appId: Constant.appID, placementId: self.placementIdTextField.text)
        }
        self.rewardVideoAd.delegate = self
        self.rewardVideoAd.load()
    }
    
    @IBAction func playVideo(_ sender: Any) {
        if (self.rewardVideoAd.expiredTimestamp <= Int(Date.init().timeIntervalSince1970)) {
            self.statusLabel.text = "广告已过期，请重新拉取"
            return
        }
        if (!self.rewardVideoAd.isAdValid) {
            self.statusLabel.text = "广告失效，请重新拉取"
            return
        }
        self.rewardVideoAd.show(fromRootViewController: self)
    }
    
    @IBAction func changeOrientation(_ sender: Any) {
        if UIInterfaceOrientationIsPortrait(UIApplication.shared.statusBarOrientation) {
            self.supportOrientation = UIInterfaceOrientation.landscapeRight
        } else {
            self.supportOrientation = UIInterfaceOrientation.portrait
        }
        UIDevice.current.setValue(self.supportOrientation.rawValue, forKey: "orientation")
    }
    
    override var shouldAutorotate: Bool
    {
        return true
    }
    
    func gdt_rewardVideoAdDidLoad(_ rewardedVideoAd: GDTRewardVideoAd!) {
        self.statusLabel.text = "广告数据加载成功"
    }
    
    func gdt_rewardVideoAdVideoDidLoad(_ rewardedVideoAd: GDTRewardVideoAd!) {
        self.statusLabel.text = "视频文件加载成功"
    }
    
    func gdt_rewardVideoAdWillVisible(_ rewardedVideoAd: GDTRewardVideoAd!) {
        print("视频播放页即将打开")
    }
    
    func gdt_rewardVideoAdDidExposed(_ rewardedVideoAd: GDTRewardVideoAd!) {
        print("广告已曝光")
    }
    
    func gdt_rewardVideoAdDidClicked(_ rewardedVideoAd: GDTRewardVideoAd!) {
        self.statusLabel.text = "广告已关闭"
        print("广告已关闭")
    }
    
    func gdt_rewardVideoAdDidClose(_ rewardedVideoAd: GDTRewardVideoAd!) {
        print("广告已点击")
    }
    
    func gdt_rewardVideoAd(_ rewardedVideoAd: GDTRewardVideoAd!, didFailWithError error: Error!) {
        let code = (error as NSError).code
        if (code == 4014) {
            print("请拉取到广告后再调用展示接口")
            self.statusLabel.text = "请拉取到广告后再调用展示接口"
        } else if (code == 4016) {
            print("应用方向与广告位支持方向不一致")
            self.statusLabel.text = "应用方向与广告位支持方向不一致"
        } else if (code == 5012) {
            print("广告已过期")
            self.statusLabel.text = "广告已过期"
        } else if (code == 4015) {
            print("广告已经播放过，请重新拉取")
            self.statusLabel.text = "广告已经播放过，请重新拉取"
        } else if (code == 5002) {
            print("视频下载失败")
            self.statusLabel.text = "视频下载失败"
        } else if (code == 5003) {
            print("视频播放失败")
            self.statusLabel.text = "视频播放失败"
        } else if (code == 5004) {
            print("没有合适的广告")
            self.statusLabel.text = "没有合适的广告"
        } else if (code == 5013) {
            print("请求太频繁，请稍后再试")
            self.statusLabel.text = "请求太频繁，请稍后再试"
        }
        print("\(error)")
    }
    
    func gdt_rewardVideoAdDidRewardEffective(_ rewardedVideoAd: GDTRewardVideoAd!) {
        print("播放达到激励条件")
    }
    
    func gdt_rewardVideoAdDidPlayFinish(_ rewardedVideoAd: GDTRewardVideoAd!) {
        print("播放达到激励条件")
    }
    
}
